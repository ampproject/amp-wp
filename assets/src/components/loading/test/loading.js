/**
 * External dependencies
 */
import { act } from 'react-dom/test-utils';
import { create } from 'react-test-renderer';

/**
 * WordPress dependencies
 */
import { render } from '@wordpress/element';

/**
 * Internal dependencies
 */
import { Loading } from '..';

let container;

describe('the Loading component', () => {
	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		document.body.removeChild(container);
		container = null;
	});

	it('matches the snapshots', () => {
		const wrapperBlock = create(<Loading />);
		const wrapperInline = create(<Loading inline={true} />);

		expect(wrapperBlock.toJSON()).toMatchSnapshot();
		expect(wrapperInline.toJSON()).toMatchSnapshot();
	});

	it('renders a loading spinner', () => {
		act(() => {
			render(<Loading />, container);
		});

		expect(
			container.querySelector('.amp-spinner-container')
		).not.toBeNull();
		expect(container.querySelector('.amp-spinner-container').tagName).toBe(
			'DIV'
		);
		expect(container.querySelector('.components-spinner')).not.toBeNull();
	});

	it('renders an inline loading spinner', () => {
		act(() => {
			render(<Loading inline={true} />, container);
		});

		expect(
			container.querySelector('.amp-spinner-container--inline')
		).not.toBeNull();
		expect(
			container.querySelector('.amp-spinner-container--inline').tagName
		).toBe('SPAN');
	});
});
