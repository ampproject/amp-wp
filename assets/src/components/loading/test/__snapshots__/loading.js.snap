// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Loading component matches the snapshots 1`] = `
<div
  className="amp-spinner-container"
>
  <svg
    className="components-spinner css-17rkb14-StyledSpinner ea4tfvq2"
    focusable="false"
    height="16"
    role="presentation"
    viewBox="0 0 100 100"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      className="css-mx768t-SpinnerTrack-commonPathProps ea4tfvq1"
      cx="50"
      cy="50"
      r="50"
      vectorEffect="non-scaling-stroke"
    />
    <path
      className="css-7c7ebc-SpinnerIndicator-commonPathProps ea4tfvq0"
      d="m 50 0 a 50 50 0 0 1 50 50"
      vectorEffect="non-scaling-stroke"
    />
  </svg>
</div>
`;

exports[`the Loading component matches the snapshots 2`] = `
<span
  className="amp-spinner-container amp-spinner-container--inline"
>
  <svg
    className="components-spinner css-17rkb14-StyledSpinner ea4tfvq2"
    focusable="false"
    height="16"
    role="presentation"
    viewBox="0 0 100 100"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      className="css-mx768t-SpinnerTrack-commonPathProps ea4tfvq1"
      cx="50"
      cy="50"
      r="50"
      vectorEffect="non-scaling-stroke"
    />
    <path
      className="css-7c7ebc-SpinnerIndicator-commonPathProps ea4tfvq0"
      d="m 50 0 a 50 50 0 0 1 50 50"
      vectorEffect="non-scaling-stroke"
    />
  </svg>
</span>
`;
